File: 49-group-anagrams/group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        vector<vector<string>> ans;
        unordered_map<string, vector<string>> wordMap;
        for(string s: strs){
            string word = s;
            sort(word.begin(), word.end());
            wordMap[word].push_back(s);
        }

        for(auto &it: wordMap){
            ans.push_back(it.second);
        }
        return ans;
    }
};

File: 49-group-anagrams/group-anagrams.js
/**
 * @param {string[]} strs
 * @return {string[][]}
 */
var groupAnagrams = function(strs) {
    const group = new Map();

    for(let s of strs){
        const word = s.split('').sort().join('');

        if(!group.has(word)){
            group.set(word, [])
        }
        group.get(word).push(s);
    }

    return Array.from(group.values());
};